src/datagrid/DataGrid.tsx

```diff
@@ -294,13 +297,26 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
     // Function to return all selected rows
     getSelectedRows = (): DataGridRow[] => {
         const {allRows} = this.state;
-        let selectedRows = new Array();
-        if (this.state.selectAll) {
-            selectedRows = allRows;
-        } else {
-            selectedRows = allRows.filter(row => row["isSelected"] === true);
+        return allRows.filter(row => row.isSelected === true);
+    };
+
+    // Function to return all selection enabled rows
+    getSelectionEnabledRows = (allRows: DataGridRow[]): DataGridRow[] => {
+        return allRows.filter(row => !row.disableRowSelection);
+    };
+
+    // Function to check if all selectable rows are selected or not
+    isAllRowsSelected = (allRows: DataGridRow[]): boolean => {
+        const rows = this.getSelectionEnabledRows(allRows);
+        if (rows && rows.length) {
+            return allTrueOnKey(rows, "isSelected");
```

Comment:
please move the "isSelected" as constant or enum, if we have multiple of there

Reply:
done